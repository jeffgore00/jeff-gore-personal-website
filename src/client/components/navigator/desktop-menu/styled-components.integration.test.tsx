/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import {
  DesktopNavMenuContainer,
  StyledLink,
  NonFirstHyperlink,
} from './styled-components';

// Can't use spyOn for this library for some reason. Get "TypeError: Cannot redefine property: useLocation"
jest.mock('react-router-dom', () => ({
  Link: ({
    to,
    children,
    className,
  }: {
    to: string;
    children: React.ReactChild;
    className: string;
  }) => (
    <div className={className} id={`link-to-${to}`}>
      {children}
    </div>
  ),
}));

describe('DesktopNavMenuContainer presentational component', () => {
  it('renders the wrapped content with the expected CSS styles and HTML attributes', () => {
    const { container } = render(
      <DesktopNavMenuContainer>
        The Algorithms That Still Matter
      </DesktopNavMenuContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('StyledLink presentational component', () => {
  it('renders the React Router <Link> with the expected CSS styles', () => {
    const { container } = render(<StyledLink to="/blog" key="Blog" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('NonFirstHyperlink presentational component', () => {
  it('renders the wrapped content with the expected CSS styles and HTML attributes', () => {
    const { container } = render(
      <NonFirstHyperlink>
        <a href="/blog">Blog</a>
      </NonFirstHyperlink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
