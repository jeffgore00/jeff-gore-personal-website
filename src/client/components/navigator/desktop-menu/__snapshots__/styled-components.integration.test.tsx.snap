// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavMenuContainer presentational component renders the wrapped content with the expected CSS styles and HTML attributes 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
  data-testid="desktop-nav-menu"
>
  The Algorithms That Still Matter
</div>
`;

exports[`NonFirstHyperlink presentational component renders the wrapped content with the expected CSS styles and HTML attributes 1`] = `
.c0 {
  margin-left: 24px;
}

<div
  class="c0"
>
  <a
    href="/blog"
  >
    Blog
  </a>
</div>
`;

exports[`StyledLink presentational component renders the React Router <Link> with the expected CSS styles 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c0:visited {
  color: black;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: purple;
}

<div
  class="c0"
  id="link-to-/blog"
/>
`;
